doc/syntax/pattern_matching.rdoc

```diff
@@ -415,6 +415,11 @@ Additionally, when matching custom classes, the expected class can be specified
   end
   #=> "matched: 1"
 
+As of Ruby 3.2, these core and library classes implement deconstruction:
```

Comment:
Don't think we should mention the version it was added here.  `doc` should describe the current version of Ruby always.  Users that want documentation for previous versions should look at the documentation generated from that version.

Reply:
My point was that for quickly-changed parts of the language it is meaningful to include the version of relevance due to two facts:
1. We not always update things at time. Say, 3.2 was released with pattern-matching docs saying "as of 3.1, it is experimental". If the user encounters this statement _with_ version of relevance included, it at least gives some clue "well, this is outdated, please ignore". Counter-example: [Object#freeze](https://docs.ruby-lang.org/en/3.2/Object.html#method-i-freeze) docs say, as of 3.2, "Objects of the following classes are always frozen: Integer, Float, Symbol." -- which is long irrelevant,  because a list is much longer. And it might be that if it would say "as of 2.7, those objects are always frozen", it would be much more informative.
2. Nowadays, users frequently navigate to docs from Google, not render them locally, and if they are looking at docs for a newer version than they have, in most cases, it is either a) what they found works or b) what they found just NoMethodError or something like this, giving clear indication "your Ruby don't have it". But for subtler features like this one (what can be used in PM), it might make sense to provide a clue that it is recently-introduced feature.

But this is not a hill I am dying on, so if you find these reasons insufficient, I will remove it.

Reply:
There are certainly benefits of listing which version added which feature.  It makes code archeology easier, for one.  The cost of doing so is you eventually end up with a bunch of clutter listing which version added each feature.  Do users today care much if features were added in Ruby 1.8 as opposed to Ruby 1.9?  Eventually removing the mentioning of old versions is a fixable problem, but it requires continuous maintenance. In general, I don't think we should create more maintenance work for ourselves.

Reply:
Removed.